### ISSUE TITLE
Confusion matrix not working for multi label image classification

### ISSUE THREAD

**Issue by danielsjf** [fastai/fastai#3257]
Please confirm you have the latest versions of fastai, fastcore, and nbdev prior to reporting a bug (delete one): YES

**Describe the bug**
I'm trying to use the confusion matrix for multi label image classifications. Unfortunately, the results are not interpreted well and it produces a matrix with only results for two labels as you can see in the image below.
![image](https://user-images.githubusercontent.com/6696142/110916199-d622f100-8318-11eb-905f-8e21b4258406.png)

I already tried to report this on the forum where they redirected me to the code. I tried to investigate this issue, but I don't understand the code well enough to make quick changes that would alleviate this issue.
https://forums.fast.ai/t/confusion-matrix-for-multicategory-not-working-as-expected/86381

**To Reproduce**
The easiest way to reproduce this is from the fastbook course.
1. Follow the course about multi category labels (course 6):
https://github.com/fastai/fastbook/blob/master/06_multicat.ipynb
2. Execute the regular code until line:
```
learn = cnn_learner(dls, resnet50, metrics=partial(accuracy_multi, thresh=0.2))
learn.fine_tune(3, base_lr=3e-3, freeze_epochs=4)
```
3. After this, add the confusion matrix code from course 2:
```
interp = ClassificationInterpretation.from_learner(learn)
interp.plot_confusion_matrix(figsize=(10,10))
```

As you can see, the matrix correctly identified all the classes (vocab), but unfortunately it puts all observations in only two categories. I expect this has something to do with the fact that one image can have multiple labels, but I do not see why most labels are empty. if I look at other results it seems that the confusion matrix is off.

**Expected behavior**
In case of 100 images with each about 5 tags, I would expect a per label result. This means that about 500 values are classified in the matrix, which is different than for the regular classification case (where 100 would appear, one for each image).

**Error with full stack trace**
No explicit error is raised.

**Additional context**
NA

**Comment by kbressem** [comment]
I think you can't use a single confusion matrix if one image has multiple labels. The confusion matrix is only valid if there is one label per image.  Let's say, you have an image with a *dog* and the model predicts *cat*, then it has confused the two. However, if you have an image with a *dog* and a *bird* but the model predicted *cat* and *horse* you can't really tell what it did confuse (*dog* with *cat* or *dog* with *bird*) and it is not possible to plot the matrix. 
However, you could break down the results into multiple matrices. Check out the [multilabel_confusion_matrix](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.multilabel_confusion_matrix.html) from sklearn, maybe this is what you want?

**Comment by swamitagupta** [comment]
I've faced the same issue. @jph00 Would you be interested in having the multilabel confusion matrix implementation in fastai? Can I work on this and raise a PR for the same?
